src/EditorFeatures/CSharpTest/CodeActions/ReplaceMethodWithProperty/ReplaceMethodWithPropertyTests.cs

```diff
@@ -0,0 +1,276 @@
+        public void TestUpdateGetSet_SetterAccessibility()
+        {
+            Test(
+@"using System; class C { public int [||]GetFoo() { } private void SetFoo(int i) { } }",
+@"using System; class C { public int Foo { get { } private set { } } }",
+index: 1);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)]
+        public void TestUpdateGetSet_ExpressionBodies()
+        {
+            Test(
+@"using System; class C { int [||]GetFoo() => 0; void SetFoo(int i) => Bar(); }",
+@"using System; class C { int Foo { get { return 0; } set { Bar(); } } }",
+index: 1);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)]
+        public void TestUpdateGetSet_GetInSetReference()
+        {
+            Test(
+@"using System; class C { int [||]GetFoo() { } void SetFoo(int i) { } void Bar() { SetFoo(GetFoo() + 1); } }",
+@"using System; class C { int Foo { get { } set { } } void Bar() { Foo = Foo + 1; } }",
+index: 1);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)]
+        public void TestUpdateGetSet_UpdateSetPrameterName()
+        {
+            Test(
+@"using System; class C { int [||]GetFoo() { } void SetFoo(int i) { v = i; } }",
+@"using System; class C { int Foo { get { } set { v = value; } } }",
+index: 1);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)]
+        public void TestUpdateGetSet_SetReferenceInSetter()
+        {
+            Test(
+@"using System; class C { int [||]GetFoo() { } void SetFoo(int i) { SetFoo(i - 1); } }",
+@"using System; class C { int Foo { get { } set { Foo = value - 1; } } }",
+index: 1);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)]
+        public void TestVirtualGetWithOverride_1()
+        {
+            Test(
+@"class C { protected virtual int [||]GetFoo() { } } class D : C { protected override int GetFoo() { } }",
+@"class C { protected virtual int Foo { get { } } } class D : C { protected override int Foo{ get { } } }",
+index: 0);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)]
+        public void TestVirtualGetWithOverride_2()
+        {
+            Test(
+@"class C { protected virtual int [||]GetFoo() { } } class D : C { protected override int GetFoo() { base.GetFoo(); } }",
+@"class C { protected virtual int Foo { get { } } } class D : C { protected override int Foo { get { base.Foo; } } }",
+index: 0);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)]
+        public void TestGetWithInterface()
+        {
+            Test(
+@"interface I { int [||]GetFoo(); } class C : I { public int GetFoo() { } }",
+@"interface I { int Foo { get; } } class C : I { public int Foo { get { } } }",
+index: 0);
+        }
```

Comment:
tests:
* if invoked on the interface implementation, it updates interface definition
* test the same but with interface definition in metadata :wink: